packages/patternfly-4/react-charts/src/components/ChartArea/ChartArea.js

```diff
@@ -0,0 +1,22 @@
+import PropTypes from 'prop-types';
+import { VictoryArea } from 'victory';
+import { default as ChartTheme } from '../ChartTheme/ChartTheme';
+
+export const propTypes = {
+  /**
+   * See TypeScript API docs: https://formidable.com/open-source/victory/docs/victory-area/
+   */
+  '': PropTypes.any
+};
+
+export default class ChartArea extends VictoryArea {
```

Comment:
We should not be extending these. It is possible that they could internally remove clasess, etc. A better, and more "React" way, would be to do something like
```jsx
const ChartArea = (props) => <VictoryArea {...props} />

ChartArea.propTypes = propTypes;
ChartArea.defaultProps = {
   ...VictoryArea.defaultProps,
   theme: ChartTheme.default,
}
```

Another option is to do something like
```jsx
const ChartArea = (props) => <VictoryArea them={ChartTheme.default} {...props} />
```
That would pass the default theme down, and allow the user to overwrite it with a theme prop.

Reply:
I tried your suggested approach with ChartArea. It renders alone, but won't render as a child of VictoryGroup (i.e., Victory's group container). Looking at the Victory source, there may be other properties being referenced when rendering children (e.g., role, expectedComponents, etc)?

Reply:
That is definitely possible. It looks like they do for sure use role quite a bit. Without digging too much I am not sure the best way to handle this, but extending and then calling super.render is not something we should be doing. there is no garuntee that these components are classes. If they ever change (could even be in a minor) it would break. Example
```jsx
function SomeChartFromVictory() {
	return <div/>;
}

class OtherChart extends SomeChartFromVictory {
  render() {
    return super.render();
  }
}
const comp = new OtherChart();
comp.render();
```
That would throw an error.

Victory does have a section for Wrapping components: https://formidable.com/open-source/victory/guides/custom-components/#wrapping-components. That might be useful to look into and could be an option. I have not had time to look into this, and may not anytime soon so I am not positive that is the best option though.

Reply:
This doesn't really apply to the ChartArea, tho -- it has no children. That would apply more to a container like ChartGroup. 

To eliminate that wrapping code from the equation, I've been testing ChartArea as a child of VictoryGroup.

Reply:
Since we're wrapping these components perhaps we can extend them for now and then iterate on this later and see if we can compose and pass props through in a way that Victory likes

Reply:
Yes, iterating on this would be my preference.

FYI, I was able to get a few components to work using Doug's approach, by adding extra properties. For example, most components required a `role` in order to render, while ChartVoronoiContainer also required `defaultEvents`. And ChartBar required `getDomain`. 

Unfortunately, I've yet to get ChartTooltip to work properly. It renders with ChartVoronoiContainer, but not with ChartPie or ChartDonut. Adding properties seems more fragile than extending because they could change.

I also noticed a new violation message with this approach that is not present when extending.

`[Violation] 'DOMContentLoaded' handler took 422ms`

Reply:
I have the components working, using Doug's suggested approach, but had to add some extra properties. For example, most components required a `role` in order to render, while ChartVoronoiContainer also required `defaultEvents`. ChartBar required `getDomain` and ChartStack needed `getChildren`.

I wanted to use Object.assign to add all properties, but that doesn't work in all cases (e.g., I still needed to add `role` & `getChildren` to ChartStack) and some inherited properties were invalid.

This seems more fragile than just extending the class? That is, what's currently a required property could change.

Reply:
This might help: https://github.com/mridgway/hoist-non-react-statics That will allow us to not worry about what static properties each of this components need. We can hoist them up to our component.

Reply:
Yes, that is exactly what was needed. Thank you!